patches/bess/0013-Use-rte_flow_create-to-do-GTPU-RSS.patch

```diff
@@ -74,45 +63,91 @@ index 7822e7ef..913dbc73 100644
 +  memset(&action_rss, 0, sizeof(action_rss));
 +  action_rss.func = RTE_ETH_HASH_FUNCTION_DEFAULT;
 +  action_rss.key_len = 0;
++  action_rss.types = rss_types;
 +
 +  actions[0].type = RTE_FLOW_ACTION_TYPE_RSS;
 +  actions[0].conf = &action_rss;
 +  actions[1].type = RTE_FLOW_ACTION_TYPE_END;
 +
-+  switch (flow_profile)
-+  {
-+  case 3:
-+    action_rss.types = ETH_RSS_IPV4 | ETH_RSS_L3_SRC_ONLY;
-+    break;
-+  case 6:
-+    action_rss.types = ETH_RSS_IPV4 | ETH_RSS_L3_DST_ONLY;
-+    patterns[2].type = RTE_FLOW_ITEM_TYPE_END;
-+    break;
-+  case 9:
-+    action_rss.types = ETH_RSS_IPV4 | ETH_RSS_L3_DST_ONLY;
-+    break;
-+  default:
-+    return CommandFailure(EINVAL, "Unknown flow profile %u", flow_profile);
-+  }
-+
-+  int ret = rte_flow_validate(port_id, &attributes, patterns, actions, &err);
-+  if(ret)
-+    return CommandFailure(EINVAL, "Port %u: Failed to validate flow profile %u %s",
++  int ret = rte_flow_validate(port_id, &attributes, items, actions, &err);
++  if (ret)
++    return CommandFailure(EINVAL,
++                          "Port %u: Failed to validate flow profile %u %s",
 +                          port_id, flow_profile, err.message);
 +
-+
-+  handle = rte_flow_create(port_id, &attributes, patterns, actions, &err);
-+  if(handle == nullptr)
-+    return CommandFailure(EINVAL, "Port %u: Failed to create flow %s",
-+                          port_id, err.message);
++  handle = rte_flow_create(port_id, &attributes, items, actions, &err);
++  if (handle == nullptr)
++    return CommandFailure(EINVAL, "Port %u: Failed to create flow %s", port_id,
++                          err.message);
 +
 +  return CommandSuccess();
 +}
++
++#define NUM_ELEMENTS(x) (sizeof(x) / sizeof((x)[0]))
++
++CommandResponse flow_create(dpdk_port_t port_id, const uint32_t &flow_profile) {
++  CommandResponse err;
++
++  rte_flow_item_type N39_NSA[] = {
++      RTE_FLOW_ITEM_TYPE_ETH, RTE_FLOW_ITEM_TYPE_IPV4, RTE_FLOW_ITEM_TYPE_UDP,
++      RTE_FLOW_ITEM_TYPE_GTPU, RTE_FLOW_ITEM_TYPE_IPV4,
++      RTE_FLOW_ITEM_TYPE_END};
++
++  rte_flow_item_type N39_SA[] = {
++      RTE_FLOW_ITEM_TYPE_ETH, RTE_FLOW_ITEM_TYPE_IPV4, RTE_FLOW_ITEM_TYPE_UDP,
++      RTE_FLOW_ITEM_TYPE_GTPU, RTE_FLOW_ITEM_TYPE_GTP_PSC,
++      RTE_FLOW_ITEM_TYPE_IPV4,
++      RTE_FLOW_ITEM_TYPE_END};
++
++  rte_flow_item_type N6[] = {
++      RTE_FLOW_ITEM_TYPE_ETH, RTE_FLOW_ITEM_TYPE_IPV4,
++      RTE_FLOW_ITEM_TYPE_END};
++
++  switch (flow_profile) {
++    uint64_t rss_types;
++    // N3 traffic with and without PDU Session container
++    case 3:
```

Comment:
can we have proper case name (using enumerators or others) instead of numerical no (3,6..)

Reply:
done